aquadoggo/src/libp2p/service.rs

```diff
@@ -0,0 +1,195 @@
+    }
+
+    // Spawn a task logging swarm events
+    let handle = tokio::spawn(async move {
+        loop {
+            match swarm.select_next_some().await {
+                SwarmEvent::BannedPeer {
+                    peer_id,
+                    endpoint: _,
+                } => info!("BannedPeer: {peer_id:?}"),
+                SwarmEvent::Behaviour(BehaviourEvent::Mdns(event)) => match event {
+                    mdns::Event::Discovered(list) => {
+                        for (peer, _multiaddr) in list {
+                            info!("mDNS discovered a new peer: {peer:?}");
+                        }
+                    }
+                    mdns::Event::Expired(list) => {
+                        for (peer, _multiaddr) in list {
+                            info!("mDNS peer has expired: {peer:?}");
+                        }
+                    }
+                },
+                SwarmEvent::Behaviour(BehaviourEvent::Ping(Event { peer, result: _ })) => {
+                    info!("Ping from: {peer:?}")
+                }
+                SwarmEvent::ConnectionClosed {
+                    peer_id,
+                    endpoint,
+                    num_established,
+                    cause,
+                } => {
+                    info!("ConnectionClosed: {peer_id:?} {endpoint:?} {num_established} {cause:?}")
+                }
+                SwarmEvent::ConnectionEstablished {
+                    peer_id,
+                    endpoint,
+                    num_established,
+                    ..
+                } => info!("ConnectionEstablished: {peer_id:?} {endpoint:?} {num_established}"),
+
+                SwarmEvent::Dialing(peer_id) => info!("Dialing: {peer_id:?}"),
+                SwarmEvent::ExpiredListenAddr {
+                    listener_id,
+                    address,
+                } => info!("ExpiredListenAddr: {listener_id:?} {address:?}"),
+
+                SwarmEvent::IncomingConnection {
+                    local_addr,
+                    send_back_addr,
+                } => info!("IncomingConnection: {local_addr:?} {send_back_addr:?}"),
+                SwarmEvent::IncomingConnectionError {
+                    local_addr,
+                    send_back_addr,
+                    error,
+                } => info!("IncomingConnectionError: {local_addr:?} {send_back_addr:?} {error:?}"),
+                SwarmEvent::ListenerClosed {
+                    listener_id,
+                    addresses,
+                    reason,
+                } => info!("ListenerClosed: {listener_id:?} {addresses:?} {reason:?}"),
+                SwarmEvent::ListenerError { listener_id, error } => {
+                    info!("ListenerError: {listener_id:?} {error:?}")
+                }
+                SwarmEvent::NewListenAddr {
+                    address,
+                    listener_id,
+                } => {
+                    info!("Listening on {address:?} {listener_id:?}");
```

Comment:
Similar here: `Listening on "/ip4/127.0.0.1/udp/48648/quic-v1" ListenerId(2898880830502420286)` (`Debug` vs. `Display`)

Reply:
Unfortunately `ListenerId` does not currently implement `Display`.

Reply:
Maybe we take it out then?

Reply:
I have removed the logging of `ListenerId` for that event and have knocked the logging of most swarm events down to `debug`. That sounds make things less noisy while still providing valuable information for debugging.